test/e2e/es_cluster_logging.go

```diff
@@ -0,0 +1,330 @@
+	// parallel executions of this test.
+	name := "synthlogger-" + string(util.NewUUID())
+	// Replace '-' characters with '_' to prevent the analyzer from breaking apart names.
+	underscoreName := strings.Replace(name, "-", "_", -1)
+
+	// podNames records the names of the synthetic logging pods that are created in the
+	// loop below.
+	var podNames []string
+	// countTo is the number of log lines emitted (and checked) for each synthetic logging pod.
+	const countTo = 100
+	// Instantiate a synthetic logger pod on each node.
+	for i, node := range nodes.Items {
+		podName := fmt.Sprintf("%s-%d", name, i)
+		_, err := c.Pods(ns).Create(&api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": name},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "synth-logger",
+						Image:   "ubuntu:14.04",
+						Command: []string{"bash", "-c", fmt.Sprintf("i=0; while ((i < %d)); do echo \"%d %s $i %s\"; i=$(($i+1)); done", countTo, i, underscoreName, podName)},
+					},
+				},
+				Host:          node.Name,
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		podNames = append(podNames, podName)
+	}
+
+	// Cleanup the pods when we are done.
+	defer func() {
+		for _, pod := range podNames {
+			if err = c.Pods(ns).Delete(pod); err != nil {
+				Logf("Failed to delete pod %s: %v", pod, err)
+			}
+		}
+	}()
+
+	// Wait for the syntehtic logging pods to finish.
+	By("Waiting for the pods to succeed.")
+	for _, pod := range podNames {
+		err = waitForPodSuccess(c, pod, "synth-logger")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Wait a bit for the log information to make it into Elasticsearch.
+	time.Sleep(30 * time.Second)
+
+	// Make several attempts to observe the logs ingested into Elasticsearch.
+	By("Checking all the log lines were ingested into Elasticsearch")
+	missing := 0
+	expected := nodeCount * countTo
+	for start := time.Now(); time.Since(start) < graceTime; time.Sleep(5 * time.Second) {
+		// Ask Elasticsearch to return all the log lines that were tagged with the underscore
+		// verison of the name. Ask for twice as many log lines as we expect to check for
+		// duplication bugs.
+		body, err = c.Get().
+			Prefix("proxy").
+			Resource("services").
+			Name("elasticsearch-logging").
+			Suffix("_search").
+			Param("q", fmt.Sprintf("log:%s", underscoreName)).
+			Param("size", strconv.Itoa(2*expected)).
+			DoRaw()
+		Expect(err).NotTo(HaveOccurred())
+
+		response, err := bodyToJSON(body)
+		if err != nil {
+			Failf("Failed to unmarshal response: %v", err)
+		}
+		hits, ok := response["hits"].(map[string]interface{})
+		if !ok {
+			Failf("response[hits] not of the expected type: %T", response["hits"])
+		}
+		totalF, ok := hits["total"].(float64)
+		if !ok {
+			Failf("hits[total] not of the expected type: %T", hits["total"])
+		}
+		total := int(totalF)
```

Comment:
If this always rounds properly, then why not use it above (`int(statusCode) != 200`) instead of doing `Abs(statusCode - 200) > 0.001`?
And if this doesn't always round properly, then it probably shouldn't be relied on here :)